cns/middlewares/k8sSwiftV2.go

```diff
@@ -125,7 +124,7 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
 			return nil, types.UnexpectedError, fmt.Errorf("failed to get pod's mtpnc from cache : %w", err).Error()
 		}
 		// Check if the MTPNC CRD is ready. If one of the fields is empty, return error
-		if mtpnc.Status.PrimaryIP == "" || mtpnc.Status.MacAddress == "" || mtpnc.Status.NCID == "" || mtpnc.Status.GatewayIP == "" {
+		if mtpnc.Status.PrimaryIP == "" || mtpnc.Status.MacAddress == "" || mtpnc.Status.NCID == "" || mtpnc.Status.GatewayIP == "" || mtpnc.Status.NICType == "" {
```

Comment:
Is it possible to put this as a method of `mtpnc`? This conditional and the one that follows is entirely duplicated and contains knowledge about what it means for the CRD to be ready. That usually makes for a good candidate for extracting a method.

Reply:
Added helper function.